compiler/src/iree/compiler/Codegen/Common/ReconcileTranslationInfo.cpp

```diff
@@ -295,6 +291,10 @@ static LogicalResult resolveWorkgroupForAll(RewriterBase &rewriter,
                               "scf.forall ops withing the function");
   }
 
+  if (!llvm::hasSingleElement(body)) {
+    return funcOp.emitOpError("unhandled function with multiple blocks");
+  }
```

Comment:
Why do we need this change?

Reply:
It's because of this issue: https://github.com/iree-org/iree/issues/18633

Reply:
I am just triaging this here. I will create separate PRs.